internal.nix

```diff
@@ -331,6 +331,9 @@ rec {
 
               ${preInstallLinkCommands}
 
+              cat package.json | jq -r '. | select(.bin != null) | .bin | values[]' | while read binTarget; do
```

Comment:
Is the `package.json` guranteed to be there?

Reply:
We could also pass the `package.json` as 2nd argument to cat and avoid the cat invocation.

Reply:
> Is the `package.json` guranteed to be there?

this is called for every npm package, these are required to have a package.json

> We could also pass the `package.json` as 2nd argument to jq

its prettier this way ; ) but yes, lets save some nanoseconds and call only jq

Reply:
> Is the package.json guranteed to be there?

package.json \*should\* always be there
and if its missing, we should throw a fatal error

currently, the script continues and throws later

i managed to hack together a broken node_modules with missing submodules ...
in that case i got 

```
> semver@5.7.1 preinstall /build/node_modules
> /build/node_modules/.hooks/preinstall
jq: error: Could not open file package.json: No such file or directory
patching script interpreter paths in .
```

but package.json should be there

```
$ tar tf /nix/store/gll14kwxsz9ggfhngafafh64rv4a02fr-semver-5.7.1.tgz
package/LICENSE
package/bin/semver
package/range.bnf
package/semver.js
package/package.json
package/CHANGELOG.md
package/README.md
```

Reply:
I also encountered a similar problem, where `jq` was complaining there was no "package.json" even though it was fine accesing it with `cat` and `ls` when I was working on #108. I think using `cat <file> | jq ...` was mitigating the problem but then I stopped encountering it after fixing a some other part of the bash code. I still have no idea why that happened/happens.

Reply:
> package.json *should* always be there
> and if its missing, we should throw a fatal error

no. there are packages without a package.json,
for example type definitions (\*.d.ts) or header files (\*.h)

https://github.com/pnpm/pnpm/issues/3728